core/src/main/java/hudson/model/Cause.java

```diff
@@ -366,9 +367,15 @@ public UserCause() {
             this.authenticationName = Jenkins.getAuthentication().getName();
         }
 
+        /**
+         * Gets user display name when possible.
+         * @return User display name.
+         *         If the User does not exist, returns its ID.
+         */
         @Exported(visibility=3)
         public String getUserName() {
-        	return User.get(authenticationName).getDisplayName();
+        	final User user = User.get(authenticationName, false, Collections.emptyMap());
```

Comment:
`getById`

Reply:
I am not 100% sure about that. That's why I decided to keep the original behavior.

Reply:
I am fairly confident the original behavior was wrong.